### PR TITLE
libvfio-user.c: use ERROR_INT()

### PR DESCRIPTION
**Pull request by jlevon**

### CODE REVIEW

File: lib/libvfio-user.c [hunk 1]
```diff
@@ -862,17 +852,13 @@ MOCK_DEFINE(exec_command)(vfu_ctx_t *vfu_ctx, struct vfio_user_header *hdr,
             *iovecs = _iovecs;
             *nr_iovecs = 2;
         } else {
-            ret = -errno;
             free(irq_info);
```
Comment (tmakatos) [inline lib/libvfio-user.c:855]: Won't `free` overwrite `errno` set by `handle_device_get_irq_info`?

**Review by tmakatos — COMMENTED** [review]
See comments inline.

File: lib/libvfio-user.c (hunk 1 above)
Comment (jlevon) [reply lib/libvfio-user.c:855]: You're technically right. I always forget that this isn't guaranteed (yet) by Linux.

https://sourceware.org/bugzilla/show_bug.cgi?id=17924

Surprised that is so recently fixed.

**Comment by tmakatos** [comment]
Is there anything new for me to review here?

**Comment by jlevon** [comment]
@tmakatos you can take a quick look at https://github.com/nutanix/libvfio-user/pull/433/files/73bdc53d027db1354ad8173316addbabc2ca98a4..8b8dd22a83390b24a8d0e78698fb27948d005032

**Review by tmakatos — APPROVED** [review]

**Pull request merged**